src/tequila/grouping/overlapping_methods.py

```diff
@@ -0,0 +1,203 @@
+import numpy as np
+from tequila.grouping.binary_utils import sorted_insertion_grouping, term_commutes_with_group
+from copy import deepcopy
+
+class OverlappingHelp:
+    '''
+    Class required for passing cov_dict and number of iterations to
+    OverlappingGroups. Eventually, this may also be used for building cov_dict from
+    approximate wavefunction if the user provides only the latter.
+    '''
+    def __init__(self, cov_dict, n_iter=5):
+        self.cov_dict = cov_dict
+        self.n_iter = n_iter
+
+class OverlappingGroups_wo_fixed:
+    '''
+    Class required for mapping OverlappingGroups onto a vector of overlapping coefficients.
+    Same as OverlappingGroups, but Pauli products corresponding to fixed coefficients are removed.
+    '''
+    def __init__(self, o_groups, o_terms, term_exists_in):
+        def exclude_fixed_coeffs(o_groups, o_terms, term_exists_in):
+            '''
+            Remove fixed coefficients from o_groups and term_exists_in.
+            These fixed coefficients are determined from other Pauli coefficients in order that
+            the condition sum_alpha c^{alpha} = 0 is satisfied.
```

Comment:
`c^{alpha} = 0` depends on how you define c^{alpha}. It's probably easiest to reference an equation in Aadi's paper